calisphere/utils.py

```diff
@@ -28,3 +28,17 @@ def json_loads_url(url_or_req):
         except urllib.error.HTTPError:
             data = {}
     return data
+
+# Escape OpenSearch `query_string` query reserved characters
+# (Solr has a smaller, overlapping subset of special characters)
+# Exceptions:
+#  - we are allowing phrase queries surrounded by ""
+#  - we are allowing wildcard expressions using *
+def query_string_escape(text):
+    for reserved in ['\\', '+', '-', '=', '&&', '||',
+                     '!', '(', ')', '{', '}',
+                     '[', ']', '^', '~', '?',
+                     "'", ':', '/']:
+        text = text.replace(reserved, f'\\{reserved}')
+
+    return text
```

Comment:
Have you tested this against Solr as well? 

Presumably it is okay to escape these characters even though Solr doesn't seem to _need_ us to escape them. Since we're doing all of our search behavior comparisons against calisphere prod's search behavior, though, I'm a bit worried about inadvertently changing the goal post.

Reply:
Yes, I did test it against Solr. We had already previously added `:` to the list of escaped characters in order to support OpenSearch ID queries with ARKs in them, so I figured it would be fine, but I also wanted to make sure.

Reply:
It looks like it wouldn't be too onerous to reinstate a separate `solr_escape` and apply that only to Solr queries.

Reply:
(Let me know if you'd like me to do that! I feel OK leaving things as they are, but understand it's somewhat going on faith.)